syntax = "proto3";

package org.couchers.json;

import "google/protobuf/empty.proto";

import "pb/google/api/annotations.proto";
import "pb/google/api/httpbody.proto";

service GIS {
  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto

  rpc GetUsers(google.protobuf.Empty) returns (google.api.HttpBody) {
    option (google.api.http) = {
      get: "/geojson/users"
    };
  }

  rpc GetCommunities(google.protobuf.Empty) returns (google.api.HttpBody) {
    option (google.api.http) = {
      get: "/geojson/communities"
    };
  }

  // Returns whether the given point is on land or not
  rpc IsOnLand(IsOnLandReq) returns (IsOnLandRes);
}

message IsOnLandReq {
  double lat = 1;
  double lng = 2;
}

message IsOnLandRes {
  bool on_land = 1;
}
